scripts/utils/generate_package_wiki.py

```diff
@@ -0,0 +1,88 @@
+from collections import defaultdict
+
+packages_by_category = defaultdict(str)
+
+
+def sort_write_wiki_content(file_path):
+    """ Writes content of packages_by_category to the wiki file
+    
+    This function retrieves rows pertaining to the key category, as a result of
+    iterating over packages_by_category.
+    Writes to the Markdown formatted string:
+    - header with the category name
+    - table with two columns: Package and Description of each of the packages
+    
+    Args:
+       file_path: path of the wiki .md file
+    """
+    wikiContent = ""
+    for category in packages_by_category.keys():
+        wikiContent += "## " + category + "\n\n"
+        wikiContent += "| Package | Description |\n"
+        wikiContent += "|---|---|\n"
+        wikiContent += packages_by_category[category] + "\n\n"
+    with open(file_path, "w", encoding="utf-8") as f:
+        f.write(wikiContent)
+
+
+def find_element_text(parent, tag_name):
+    """Retrieves the text content of a specified XML element.
+
+    This function searches for a child element with the given tag name within the
+    provided parent element. It handles XML namespaces by using a wildcard.
+
+    Args:
+        parent: The parent XML element.
+        tag_name: The name of the child element to find.
+
+    Returns:
+        The text content of the found element, or None if the element is not found.
+    """
+    tag = parent.find(f"{{*}}{tag_name}")
+    if tag is None:
+        return None
+    return tag.text
+
+
+def process_packages_directory(packages_dir):
+    """ Obtains the package name, description and category from a directory
+    
+    This function parses all the nuspec files in the 'packages' directory structure.
+    It saves into packages_by_category[category] a line containing the package
+    name and the description separated by |
+    
+    Args:
+        packages: directory where all the packages reside
+    """
+    if not os.path.isdir(packages_dir):
+        raise FileNotFoundError(f"Packages directory not found: {packages_dir}")
+
+    for nuspec_path in pathlib.Path(packages_dir).glob("**/*.nuspec"):
+        nuspec_tree = ET.parse(nuspec_path)
+        nuspec_metadata = nuspec_tree.find("{*}metadata")
+        if nuspec_metadata is not None:
+            category = find_element_text(nuspec_metadata, "tags")
+            # we are only interested in the packages with a category
+            # some packages that assist with the istallation (such as common.vm) do
+            # not contain a category
+            if not category:
+                continue
+            description = find_element_text(nuspec_metadata, "description")
+            package = find_element_text(nuspec_metadata, "id")
+            """packages_by_category is a dictionary of str where each package and description
+            are appended formatted in Markdown"""
+            packages_by_category[category] += "|" + package + "|" + description + "|\n"
```

Comment:
Adding some spaces makes the markdown nicer to read:
```suggestion
            packages_by_category[category] += "| " + package + " | " + description + " |\n"
```